lib/lib.go

```diff
@@ -252,3 +292,108 @@ func (k *Khan) RetrieveClan(ctx context.Context, clanID string) (*Clan, error) {
 	err = json.Unmarshal(body, &clan)
 	return &clan, err
 }
+
+// ApplyForMembership calls apply for membership route on khan
+func (k *Khan) ApplyForMembership(
+	ctx context.Context,
+	payload *ApplicationPayload,
+) (*ClanApplyResult, error) {
+	route := k.buildApplyForMembershipURL(payload.ClanID)
+	body, err := k.sendTo(ctx, "POST", route, payload)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var application ClanApplyResult
+	err = json.Unmarshal(body, &application)
+	return &application, err
+}
+
+// InviteForMembership invites a clan member to join clan
+func (k *Khan) InviteForMembership(
+	ctx context.Context,
+	payload *InvitationPayload,
+) error {
+	route := k.buildInviteForMembershipURL(payload.ClanID)
+	_, err := k.sendTo(ctx, "POST", route, payload)
+	return err
```

Comment:
Are you sure we won't need the reply for this and other methods such as `ApproveDenyMembershipApplication`? I would rather return the response everywhere and the client application can choose to ignore it if necessary.

Reply:
You're right! This also reduces the possibility of breaking changes. I will change it